testsuite/flattening/modelica/scodeinst/ArrayConnect3.mo

```diff
@@ -0,0 +1,120 @@
+// name: ArrayConnect3
+// keywords:
+// status: correct
+// cflags: -d=newInst,arrayConnect,-nfScalarize
+//
+
+connector C
+  Real e;
+  flow Real f;
+end C;
+
+model A
+  C p;
+  C n;
+end A;
+
+model Cell
+  C l, r, u, d;
+end Cell;
+
+model ArrayConnect3
+  parameter Integer N = 1000;
+  parameter Integer M = 100;
+  A S;
+  Cell cells[N, M];
+equation
+  for i in 1:N-1, j in 1:M-1 loop
+    connect(cells[i, j].r, cells[i, j+1].l);
+    connect(cells[i, j].d, cells[i+1, j].u);
+  end for;
+
+  for i in 1:N loop
+    connect(cells[i, M].r, cells[i, 1].l);
+  end for;
+
+  for j in 1:M loop
+    connect(cells[1, j].u, S.p);
+    connect(cells[N, j].d, S.n);
+  end for;
+end ArrayConnect3;
+
+// Result:
+// class ArrayConnect3
+//   final parameter Integer N = 1000;
+//   final parameter Integer M = 100;
+//   Real S.p.e;
+//   Real S.p.f;
+//   Real S.n.e;
+//   Real S.n.f;
+//   Real[1000, 100] cells.d.f;
+//   Real[1000, 100] cells.d.e;
+//   Real[1000, 100] cells.u.f;
+//   Real[1000, 100] cells.u.e;
+//   Real[1000, 100] cells.r.f;
+//   Real[1000, 100] cells.r.e;
+//   Real[1000, 100] cells.l.f;
+//   Real[1000, 100] cells.l.e;
+// equation
+//   for $i1 in 1:999 loop
+//     for $i2 in 2:100 loop
+//       cells[$i1,$i2].l.e = cells[$i1,$i1 - 1].r.e;
```

Comment:
I'm not entirely sure what went wrong here, but the iterators are not correct on the right side of the equation.
Should be `cells[$i1,$i2 - 1].r.e` right?
Also the body of the next for loop is wrong.

Reply:
You're right, but unfortunately it's the same result that ModelicaCC gives. So there seems to be something wrong with the algorithm, or at least the way it's implemented in ModelicaCC. It seems to always use only the first iterator on the rhs when there's two dimensions involved, but I'm not sure why.

Reply:
This might be the issue:
https://github.com/OpenModelica/OpenModelica/blob/f86be0fb287dc9b92e97190c6f10775cb49726ad/OMCompiler/Compiler/NFFrontEnd/NFArrayConnections.mo#L1019
I assume this should actually be `iterators[i]` instead.